### PR TITLE
[DOCS] Adds anomaly job health alert type docs

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds documentation about the anomaly detection job health alert type to the `Generating alerts for anomaly detection jobs` page.

### Preview

[Generating alerts for anomaly detection jobs](https://elasticsearch_76659.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-configuring-alerts.html)

### CODE REVIEW

**Comment by szabosteve** [comment]
@elasticmachine test this please

File: docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc [hunk 1]
```diff
@@ -5,42 +5,57 @@
...
 {kibana-ref}/alerting-getting-started.html#alerting-getting-started[Alerting].
 
+The following {ml} alerts are available:
 
-[[creating-anomaly-alert-rules]]
+* {anomaly-detect-cap} alert: checks if the {anomaly-job} results contain
+  anomalies that match the rule conditions.
+* {anomaly-jobs-cap} health: monitors job health and alerts if an operational 
+  issue occured that may prevent the job from detectiong anomalies. Recommended 
+  to use for jobs with critical importance.
```
Comment (darnautov) [inline docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc:22]: maybe worth mentioning jobs that have associated alerting rules of `{anomaly-detect-cap}`

File: docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc [hunk 2]
```diff
@@ -72,14 +87,54 @@ the sample results by providing a valid interval for your data. The generated
...
+As the last step in the rule creation process, 
+<<defining-actions, define the actions>> to take in case the configured criteria 
+are met.
+
+
+[[creating-anomaly-jobs-health-rules]]
+=== {anomaly-jobs-cap} health
+
+Select the {anomaly-job} or the group of {anomaly-jobs} that is checked against 
+the rule. If you assign additional jobs to the group, the new jobs are
```
Comment (darnautov) [inline docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc:99]: Great you've mentioned that! Please also add the same for the special selector `*` that allows monitoring of all existing jobs, even those created _after_ the rule instance creation.

File: docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc (hunk 1 above)
Comment (szabosteve) [reply docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc]: Further clarified in b25c915.

File: docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc (hunk 2 above)
Comment (szabosteve) [reply docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc:100]: Added in b25c915.

**Review by lcawl — APPROVED** [review]
Some more minor suggestions, otherwise LGTM

File: docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc [hunk 3]
```diff
@@ -101,3 +172,160 @@ The name of an alert is always the same as the job ID of the associated
...
+.Properties of `context.results` 
+[%collapsible%open]
+====
+`job_id`:::
+The job identifier.
+
+`memory_status`:::
+The status of the mathematical model. It can have one of the following values:
+
+* `ok`: The models stayed below the configured value.
```
Comment (darnautov) [inline docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc:267]: only `soft_limit` and `hard_limit` values are possible, we should remove this one 
```suggestion
```

File: docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc [hunk 4]
```diff
@@ -101,3 +172,160 @@ The name of an alert is always the same as the job ID of the associated
...
+  further category examples will be stored.
+* `hard_limit`: The model used more space than the configured memory limit. As a 
+  result, not all incoming data was processed.
+
+`model_bytes_exceeded`:::
+The number of bytes over the high limit for memory usage at the last allocation 
+failure.
+
+`log_time`:::
+The timestamp of the model size statistics according to server time.
```
Comment (darnautov) [inline docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc:279]: not sure if we should mention that time formatting based on the Kibana settings is applied

File: docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc [hunk 5]
```diff
@@ -101,3 +172,160 @@ The name of an alert is always the same as the job ID of the associated
...
+
+`peak_model_bytes`:::
```
Comment (darnautov) [inline docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc:281]: my bad I missed the following fields in the tech design doc: `model_bytes` and `model_bytes_memory_limit`. Could you please add them?

File: docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc [hunk 6]
```diff
@@ -101,3 +172,160 @@ The name of an alert is always the same as the job ID of the associated
...
+Contains the following properties:
++
+.Properties of `context.results`
+[%collapsible%open]
+====
+`annotation`:::
+The annotation corresponding to the data delay in the job.
+
+`end_timestamp`:::
+Timestamp of the latest finalized buckets with missing documents.
```
Comment (darnautov) [inline docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc:300]: not sure if we should mention that time formatting based on the Kibana settings is applied

File: docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc [hunk 7]
```diff
@@ -5,42 +5,62 @@
...
-Select the {anomaly-job} or the group of {anomaly-jobs} that is checked against 
-the rule. If you assign additional jobs to the group, the new jobs are 
-automatically checked the next time the conditions are checked.
+image::images/ml-rule.jpg["Creating a new machine learning rule"]
+
+
+[[creating-anomaly-alert-rules]]
+=== {anomaly-detect-cap} alert
+
+Select the job or the group that the rule applies to. If you assign more jobs to
```
Comment (darnautov) [inline docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc:59]: For anomaly detection alerting rule we only support single job selection at the moment

File: docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc [hunk 8]
```diff
@@ -72,14 +92,62 @@ the sample results by providing a valid interval for your data. The generated
...
+[[creating-anomaly-jobs-health-rules]]
+=== {anomaly-jobs-cap} health
+
+Select the job or group that
+the rule applies to. If you assign more jobs to the group, they are
+included the next time the rule conditions are checked.
+
+You can also use a special character (`*`) to apply the rule to all your jobs. 
+Jobs created after the rule are automatically included. If you have many jobs, 
+you might receive numerous notifications. To avoid this situation, you can
```
Comment (darnautov) [inline docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc:109]: We group alert instances by the health check type, so this sentence might be misleading.

File: docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc [hunk 9]
```diff
@@ -101,3 +172,160 @@ The name of an alert is always the same as the job ID of the associated
...
+The bucket timestamp of the anomaly.
+
+`context`.`timestampIso8601`::
+The bucket timestamp of the anomaly in ISO8601 format.
+
+`context`.`topInfluencers`::
+The list of top influencers.
+
+`context`.`topRecords`::
+The list of top records.
```
Comment (darnautov) [inline docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc:214]: I wonder if we should include properties of the influencers and records 🤔

Comment (szabosteve) [reply docs/reference/ml/anomaly-detection/ml-configuring-alerts.asciidoc:214]: @darnautov Can you provide a list of properties?

**Comment by szabosteve** [comment]
Review comments are addressed via f944a96.

**Review by darnautov — APPROVED** [review]
LGTM! 🚀

**Pull request merged**